onnxruntime/core/framework/session_state.h

```diff
@@ -70,7 +80,7 @@ class SessionState {
   * Adds an initialized tensor (weight) so that it can be used by the
   * execution frame to setup the appropriate MLValue vectors.
   */
-  void AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue);
+  Status AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue, const OrtDeleter& d);
```

Comment:
>const OrtDeleter& d [](start = 73, length = 19)

Seems like the OrtDeleter could be optional given we only use it if d.f is not null.

Reply:
So, are  you suggesting changing the const OrtCallback& to 'const OrtCallback*' or adding a new function? Or how to make it better?



---
In reply to: [261363290](https://github.com/Microsoft/onnxruntime/pull/520#discussion_r261363290) [](ancestors = 261363290)

Reply:
Just making it a pointer. Reference says it must be passed. Pointer says it may. If we ignore it when d.f is null, we probably don't need to force someone to pass it.